test/rp66.cpp

```diff
@@ -0,0 +1,484 @@
+
+    CHECK(bytes_read == 10);
+    CHECK(err == LFP_EOF);
+    CHECK_THAT(out, Equals(expected));
+
+    lfp_close(rp66);
+}
+
+TEST_CASE(
+    "Wrong format-version is fatal"
+    "[visible envelope][rp66]") {
+
+    const auto file = std::vector< unsigned char > {
+        /* First Visible Envelope */
+        0x00, 0x06,
+        0xFF, 0x01,
+        /* Body */
+        0x01, 0x02,
+        /* Second Visible Envelope */
+        0x00, 0x06,
+        0xFE, 0x01,
+        /* Body */
+        0x09, 0x0A,
+    };
+
+    auto* mem = lfp_memfile_openwith(file.data(), file.size());
+    auto* rp66 = lfp_rp66_open(mem);
+
+    auto out = std::vector< unsigned char >(4, 0xFF);
+    std::int64_t bytes_read = -1;
+    const auto err = lfp_readinto(rp66, out.data(), 4, &bytes_read);
+    CHECK(err == LFP_PROTOCOL_FATAL_ERROR);
+
+    lfp_close(rp66);
+}
+
+TEST_CASE_METHOD(
+    random_rp66,
+    "Visible Envelope: A file can be read in a single read",
+    "[visible envelope][rp66]") {
+    const auto records = GENERATE(1, 2, 3, 5, 8, 13);
+    make(records);
+
+    std::int64_t nread = 0;
+    const auto err = lfp_readinto(f, out.data(), out.size(), &nread);
+
+    CHECK(err == LFP_OK);
+    CHECK(nread == expected.size());
+    CHECK_THAT(out, Equals(expected));
+}
+
+TEST_CASE_METHOD(
+    random_rp66,
+    "Visible Envelope: A file can be read in multiple, smaller reads",
+    "[visible envelope][rp66]") {
+    const auto records = GENERATE(1, 2, 3, 5, 8, 13);
+    make(records);
+
+    test_split_read(this);
+}
+
+TEST_CASE_METHOD(
+    random_rp66,
+    "Visible Envelope: single seek matches underlying handle",
+    "[visible envelope][rp66]") {
+    const auto real_size = size;
+    auto records = GENERATE(1, 2, 3, 5, 8, 13);
+    make(records);
+
+    const auto n = GENERATE_COPY(take(1, random(0, real_size - 1)));
+    auto err = lfp_seek(f, n);
+    CHECK(err == LFP_OK);
+
+    const auto remaining = real_size - n;
+    out.resize(remaining);
+    auto memout = out;
+
+    std::int64_t nread = 0;
+    err = lfp_readinto(f, out.data(), out.size(), &nread);
+    CHECK(err == LFP_OK);
+
+    auto memerr = lfp_seek(mem, n);
+    CHECK(memerr == LFP_OK);
+    std::int64_t memnread = 0;
+    memerr = lfp_readinto(mem, memout.data(), memout.size(), &memnread);
+    CHECK(memerr == LFP_OK);
+
+    CHECK(nread == memnread);
+    CHECK(nread == out.size());
+    CHECK_THAT(out, Equals(memout));
+}
+
+TEST_CASE_METHOD(
+    random_rp66,
+    "Visible Envelope: multiple seeks and tells match underlying handle",
+    "[visible envelope][rp66]") {
+    const auto real_size = size;
+    const auto records = GENERATE(1, 2, 3, 5, 8, 13);
+    make(records);
+
+    for (int i = 0; i < 4; ++i) {
+        const auto n = GENERATE_COPY(take(1, random(0, real_size - 1)));
+        auto err = lfp_seek(f, n);
+        CHECK(err == LFP_OK);
+
+        auto memerr = lfp_seek(mem, n);
+        CHECK(memerr == LFP_OK);
+
+        std::int64_t tape_tell;
```

Comment:
If you don't want to move it out, at least rename. rp66 has no idea about any tape